src/mongocxx/CMakeLists.txt

```diff
@@ -220,41 +213,47 @@ install(FILES
     COMPONENT dev
 )
 
-install(TARGETS
-    mongocxx
-    RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT runtime
-    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT runtime
-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT dev
-)
-
 set(PACKAGE_INCLUDE_INSTALL_DIRS ${MONGOCXX_HEADER_INSTALL_DIR})
 set(PACKAGE_LIBRARY_INSTALL_DIRS ${CMAKE_INSTALL_LIBDIR})
 set(PACKAGE_LIBRARIES mongocxx)
 
 include(CMakePackageConfigHelpers)
 
-if (BUILD_SHARED_LIBS)
-    set(PKG "libmongocxx")
-else()
-    set(PKG "libmongocxx-static")
-endif()
+function(mongocxx_install TARGET)
```

Comment:
Apply a similar pattern here, as I suggested with `bsoncxx_add_library` and `bsoncxx_install`.